---
layout: page
title: More Information
permalink: /about/
---

### Rationale
At the most basic level, orthologous genes are genes which arise from speciation, for example, Human Talin, and Mouse Talin. Identification and accuracy of genes classed as orthologs is key for comparative genomic approaches.  Approximately 30 independent algorithms exist to classify orthologous genes, however, there is no consensus model. Some algorithms are based on species and gene phylogeny, while others are use only sequence comparison.  Phylogeny based approaches have the advantage of tracking genes through realistic evolutionary paths, however tend to be computationally intensive, and subject to error from misconstructed gene trees.  BLAST approaches are much faster, generally using an all-by-all blast of multispecies proteomes in order to generate best matches, however lack the added power of phylogenetic information.  

There is often no clear answer for which database to use to generate lists of orthologs for a comparative project.  Confounding this problem is the lack of a method to easily compare results from different databases.  Additionally, precomputed data available from the databases uses various gene naming methods, or different releases of the same database with non-overlapping gene names.  The Orthology Database Connector attempts to inform research using orthologs by standardizing and aligning ortholog groups from ten separate ortholog grouping algorithms.  Additionally, the number of databases which call a gene as an ortholog may be used as a proxy for confidence of that gene's assignment of orthology to the reference gene.  


### Ortholog database details


http://www.uniprot.org/database/?query=*&fil=category%3A%22Phylogenomic+databases%22





Every ortholog database except 
http://www.uniprot.org/uniprot/?query=database%3A%28type%3Ako%29&sort=score







### Contact me

claire.mcwhite (at) utexas.edu
