---
layout: default
title: Home
---

<div class="Instructions">
<H5>Search for an ortholog table</H5>
    <p>
    <b></b>Enter a  human proteome Uniprot Accession, ex. P00395</b>
    </p>
</div> 


<script type="text/javascript">
function changeText3(){
    var userInput = document.getElementById('userInput').value;
    var lnk = document.getElementById('lnk');
    lnk.href = "https://github.com/clairemcwhite/OrthologLinking/blob/master/" + userInput +".csv";
    lnk.innerHTML = lnk.href;
    window.open("https://github.com/clairemcwhite/OrthologLinking/blob/master/" + userInput +".csv");
}
</script>

<input type='text' id='userInput' value='' />
<input type='button' onclick='changeText3()' value='Go'/>
<p>
Link to table : <a href="" id=lnk></a> <br>
</p>
<br>

<HR>

<div class="More">
<h5>About</h5>
    <p>The Ortholog Database Connector displays multispecies orthologs to a reference human gene according to ten diverse ortholog sorting algorithms.</p>
    </p>
     <p>
    Tables indexed by UniProt Accession are currently available for the <a href="https://raw.githubusercontent.com/clairemcwhite/clairemcwhite.github.io/master/humanProteomeDetails.tsv">20,188 human protein coding genes</a> from the 2015 release of the UNIPROT reviewed human proteome ("UP000005640").  

    <p>
    
    Tables are created by searching for a gene within ortholog databases, and listing genes named as orthologs to it according to each database.  Table rows contain putative orthologs, labeled by Uniprot Accession, Uniprot Gene Name, and number of databases where that gene was called as an ortholog to the query gene.  Table column headers contain the database name and the ortholog group ID from that database. A gene which is called as an ortholog by a database will have a "1" in that database's column.   
    </p>    
   
</div> 

<div class="currentdatabases">

Ortholog groups are pulled from:
<ul>
  
  <li>KEGG Orthology (KO) release 74</li>
  <li>EnsemblCompara Gene Trees - Ensembl Genome Browser version 79</li>
  <li>OMA groups 2014 release</li>
  <li>TreeFam release 9</li>
  <li>OrthoDB version 8</li>
  <li>EggNOG version 4</li>
  <li>HOGENOM release 06</li>
  <li>HOVERGEN release 49</li>
  <li>COG 2014 update n</li>
  <li>PhylomeDB version 4 - Human Phylome</li>
</ul>
    
</div>
<HR>
<div class="More">

    <p>
        

    <h5>Table viewing instructions</h5>
    Github initially renders a ".csv" file as an interactive table.  To save a table, click the "Raw" icon, and then save the page.  The <a href="https://github.com/clairemcwhite/OrthologLinking">entire collection of tables</a> can be downloaded as a zip file. 
        

    </p>
</div>     
    


<div class="posts">
  {% for post in paginator.posts %}
  <article class="post">
    <h1 class="post-title">
      <a href="{{ site.baseurl }}{{ post.url }}">
        {{ post.title }}
      </a>
    </h1>

    <time datetime="{{ post.date | date_to_xmlschema }}" class="post-date">{{ post.date | date_to_string }}</time>

    {{ post.content }}
  </article>
  {% endfor %}
</div>

