---
title: "Untitled"
author: "Claire D. McWhite"
date: "October 18, 2018"
output: html_document
---

```{r}
library(tidyverse)
library(readxl)
library(cowplot)
library(viridis)
palette <- c("#0072B2","#E69F00","#111111")
colors_key <- c( "Blue", "Orange","Black")
```

```{r}
#Already SumScale normalized by dboutz
tmt1 <- read_excel("Liver_Total_TMT1_PSMs.xlsx")
tmt2 <- read_excel("Liver_Total_TMT2_PSMs.xlsx")

# Get rid of extra columns and tidy two experiments
tmt1_sel <- tmt1 %>% select(`Annotated Sequence`, `Protein Descriptions`, contains("Norm ")) 
tmt1_sel <- tmt1_sel  %>% select(`Annotated Sequence`, `Protein Descriptions`, contains("Norm ")) %>%
      rename(peptide = `Annotated Sequence`, descr = `Protein Descriptions`) %>%
      gather(iso, norm_abundance, -peptide, -descr) 
tmt1_sel$tmt <- "tmt1" 

tmt2_sel <- tmt2 %>% select(`Annotated Sequence`, `Protein Descriptions`, contains("Norm "))
tmt2_sel <- tmt2_sel  %>% select(`Annotated Sequence`, `Protein Descriptions`, contains("Norm ")) %>%
      rename(peptide = `Annotated Sequence`, descr = `Protein Descriptions`) %>%
      gather(iso, norm_abundance, -peptide, -descr) 
tmt2_sel$tmt <- "tmt2"

tmts <- rbind(tmt1_sel, tmt2_sel) %>%
      separate(iso, into = c("tmp", "condition", "iso"), sep = " ") %>%
      separate(condition, into = c("condition", "sampnum"), sep = 1) %>%
      mutate(condition = case_when(condition =="C" ~ "control",
                                   condition == "R" ~ "treated"))


#"Supplementary Table 3. Proteome data analysis. Only peptides with TMT reporter ion signal intensities for all ten samples were used for further bioinformatics analysis. Any duplicate peptides were removed before the data was SumScale normalised. The two normalised data files were then merged together to give one 10vs10 file comparison. Any peptides which did not have intensity values in all twenty TMT reporter ion channels were filtered out and median values were taken of the control and treated samples respectively.A total of 1906 peptides were quantified across all liver samples. Pairwise non-parametric Mann-Whitney U tests were performed and a p-value was attributed to each of the 1906 peptides s. The resulting p-values were adjusted by the Benjamini-Hochberg multi-test adjustment method for the high number of comparisons."


#CDM It doesn't seem like median values of 10 vs.10 were used. Calculating both, but mean matches paper fold changes much better (see last code block)

#List of peptides that were seen in both conditions (check this is right)
tmts_kept <- tmts %>% select(peptide, tmt, condition) %>%
      unique %>%
      group_by(peptide) %>%
      tally()  %>%
      filter(n ==4)
 

#Use median to calculate fold change between conditions
tmts_tidy_cond_med <- tmts %>% 
       filter(peptide %in% tmts_kept$peptide) %>%
       group_by(peptide, condition)  %>%
          mutate(med = median(norm_abundance)) %>%  #MEAN
          ungroup %>% 
       select(peptide, med, condition, descr) %>%
              unique %>%
              spread(condition, med) %>%
              mutate(fc= treated/control)  %>%
              mutate(log2fc = log2(fc)) %>%
       arrange(desc(fc))

#Use mean to calculate fold change between conditions
tmts_tidy_cond_mn <- tmts %>% 
       filter(peptide %in% tmts_kept$peptide) %>%
       group_by(peptide, condition)  %>%
          mutate(mn = mean(norm_abundance)) %>%  #MEAN
          ungroup %>% 
       select(peptide, mn, condition, descr) %>%
              unique %>%
              spread(condition, mn) %>%
              mutate(fc= treated/control)  %>%
              mutate(log2fc = log2(fc)) %>%
       arrange(desc(fc))

#Use median to calculate fold change between experiments
tmts_tidy_exp_med <- tmts %>% 
       filter(peptide %in% tmts_kept$peptide) %>%
       group_by(peptide, tmt)  %>%
          mutate(med = median(norm_abundance)) %>% 
          ungroup %>% 
       select(peptide, med,tmt, descr) %>%
              unique %>%
              spread(tmt, med) %>%
              mutate(fc= tmt2/tmt1)  %>%
              mutate(log2fc = log2(fc)) %>%
       arrange(desc(fc))

#Use mean to calculate fold change between experiments
tmts_tidy_exp_mn <- tmts %>% 
       filter(peptide %in% tmts_kept$peptide) %>%
       group_by(peptide, tmt)  %>%
          mutate(mn = mean(norm_abundance)) %>% 
          ungroup %>% 
       select(peptide, mn,tmt, descr) %>%
              unique %>%
              spread(tmt, mn) %>%
              mutate(fc= tmt2/tmt1)  %>%
              mutate(log2fc = log2(fc)) %>%
       arrange(desc(fc))

#Combine everything
tmts_calcs_med <- rbind(
    
    mutate(select(tmts_tidy_exp_med, peptide, fc, log2fc), comparison = "tmt1 vs tmt2"),
    mutate(select(tmts_tidy_cond_med, peptide, fc, log2fc), comparison = "control vs roundup"))

tmts_calcs_mn <- rbind(
    
    mutate(select(tmts_tidy_exp_mn, peptide, fc, log2fc), comparison = "tmt1 vs tmt2"),
    mutate(select(tmts_tidy_cond_mn, peptide, fc, log2fc), comparison = "control vs treated"))

#Decided on mean based on match to paper fold changes. Conflicts with methods
#Plot fold changed between conditions and between experiments
tmt_plot <- tmts_calcs_mn %>% ggplot(aes(fct_reorder(peptide, fc), y = log2fc, group = comparison, color = comparison)) +
       geom_point(alpha = 0.25) +
      theme(axis.text.x = element_blank(), legend.position = "none", axis.ticks.x = element_blank()) +
      scale_color_manual(values = palette) +
      xlab("Peptide") +
      ylab("Log2 Fold Change") +
      background_grid(major = "y") +
      facet_wrap(~comparison, ncol = 2) +
     NULL
tmt_plot %>% ggsave("conditions_vs_replicates_plot.png", ., width = 5, height = 5)
```


Correlate each sample with all other samples - Dan's heatmap but in R
```{r}
tmts_wide <- tmts %>% 
     mutate(sample_id = paste(tmt, iso, sep = ".")) %>%
     select(peptide, norm_abundance, sample_id) %>% group_by(peptide, sample_id) %>%
        summarize(med_norm_abundance = median(norm_abundance))  %>%
       ungroup %>%
     spread(sample_id, med_norm_abundance) 

tmts_wide_dat <- tmts_wide  %>%
    select(-peptide) %>%
    na.omit()

corrs <- round(cor(tmts_wide_dat ),2) %>% data.frame()
corrs$cond1 <- row.names(corrs)

tmt_corrplot <- corrs %>% gather(cond2, `Pearson\ncorr`, -cond1) %>%
    ggplot(aes(x = cond1, y = fct_rev(cond2), fill = `Pearson\ncorr`, value = `Pearson\ncorr`)) + 
    geom_tile() +
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust =1),
          axis.title = element_blank(),
          legend.position = "top",
          legend.key.width=unit(3,"line")) + 
    scale_fill_viridis(guide = guide_colorbar(reverse = TRUE) )


```

#Scramble condition labels and get fold changes
Assign each peptide to a random conditions to get an idea of base-line apparent fold changes for peptides.

```{r}
tmts_scramble <- tmts
tmts_scramble$condition <- sample(tmts$condition)
tmts_scramble <- tmts_scramble %>%
      mutate(tmt = paste( tmt, "random conditions",sep = " "))
tmts_w_rand <- rbind(tmts, tmts_scramble)

tmts_tidy_cond_mn_singleexp <- tmts_w_rand %>% 
       filter(peptide %in% tmts_kept$peptide) %>%
       group_by(peptide, condition, tmt)  %>%
          mutate(mn = mean(norm_abundance)) %>%  #MEAN
          ungroup %>% 
       select(peptide, mn, condition, descr, tmt) %>%
              unique %>%
              spread(condition, mn) %>%
              mutate(fc= treated/control)  %>%
              mutate(log2fc = log2(fc)) %>%
       arrange(desc(fc))

tmts_randplot <-tmts_tidy_cond_mn_singleexp  %>% ggplot(aes(fct_reorder(peptide, fc), y = log2fc, group = tmt, color = tmt)) +
       geom_point(alpha = 0.25) +
      theme(axis.text.x = element_blank(), legend.position = "none", axis.ticks.x = element_blank()) +
      scale_color_manual(values = c("#01a4ff","#0072B2", "#ffc749","#E69F00")) +
      xlab("Peptide") +
      ylab("Log2 Fold Change treated/control") +
      background_grid(major = "y") +
      facet_wrap(~tmt, ncol = 2, scales = "free_x") +
     NULL


```  

Combine to a figure
```{r}
a <- plot_grid(tmt_plot, tmt_corrplot, labels = c("A", "B"))
b <- plot_grid(tmts_randplot, NULL, rel_widths = c(1, 1),  labels = c("C", NULL))
plot_grid(a,b,ncol = 1) %>% ggsave("rat_exp_fc.png", .,device = "png", width = 10, height = 9)
plot_grid(a,b,ncol = 1) %>% ggsave("rat_exp_fc.pdf", .,device = "pdf", width = 10, height = 9)

```

Compare to paper fold changes
(haven't added in the relatively few 
So not quite exact match yet
```{r}
ppr <- read_csv("srep39328-s4.csv") 

paper_plot <-ppr %>% ggplot(aes(x = fct_reorder(Filename,`Fold Change`) , y = `Fold Change`)) + 
    geom_point()+
    theme(axis.text.x = element_blank())


tmts_tidy_cond_flip_med <- tmts_tidy_cond_med %>% #filter(abs(fc) >0.95) %>%
     mutate(flipped_fc = case_when(fc > 1 ~ fc, 
                                   fc < 1 ~ -1/fc)) 

recalc_plot_med <-  tmts_tidy_cond_flip_med %>%  ggplot(aes(x = fct_reorder(peptide,flipped_fc) , y = flipped_fc)) + 
    geom_point() +
    theme(axis.text.x = element_blank())
     
     
tmts_tidy_cond_flip_mn <- tmts_tidy_cond_mn %>% #filter(abs(fc) >0.95) %>%
     mutate(flipped_fc = case_when(fc > 1 ~ fc, 
                                   fc < 1 ~ -1/fc)) 

recalc_plot_mn <-  tmts_tidy_cond_flip_mn %>%  ggplot(aes(x = fct_reorder(peptide,flipped_fc) , y = flipped_fc)) +
    geom_point() +
    theme(axis.text.x = element_blank())



plot_grid(paper_plot, recalc_plot_med, recalc_plot_mn)
     
     
corr_w_old_med_plot <- tmts_tidy_cond_flip_med %>% #filter(abs(fc) >0.95) %>%
     mutate(peptide = str_replace(peptide,"^....","")) %>%
     mutate(peptide = str_replace(peptide,"....$","")) %>%
    left_join(ppr, by= c("peptide" = "Filename")) %>% 
    select(peptide, descr, Gene, Name, Uniprot, flipped_fc, `Fold Change`) %>%
    ggplot(aes(x = `Fold Change`, y = flipped_fc)) + 
    geom_point(alpha = 0.2)

corr_w_old_mn_plot <- tmts_tidy_cond_flip_mn %>% #filter(abs(fc) >0.95) %>%
     mutate(peptide = str_replace(peptide,"^....","")) %>%
     mutate(peptide = str_replace(peptide,"....$","")) %>%
    left_join(ppr, by= c("peptide" = "Filename")) %>% 
    select(peptide, descr, Gene, Name, Uniprot, flipped_fc, `Fold Change`) %>%
    ggplot(aes(x = `Fold Change`, y = flipped_fc)) + 
    geom_point(alpha = 0.2)

plot_grid(corr_w_old_med_plot, corr_w_old_mn_plot)

```


